### PR TITLE
Makefile / .gitignore

### PR DESCRIPTION
**Pull request by snim2**
This relates to the comments on Issue #42 about the possibility of automatically generated files becoming out of sync with the Markdown text of the manuscript.

Changes in this PR are:
1. Added a `make clean` target which deletes automatically generated files.
2. Removed automatically generated files from the repository (`manuscript.aux`, etc.).
3. Added a simple `.gitignore` file which ignores the auto-generated files, to prevent them being accidentally committed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+manuscript/manuscript.aux
+manuscript/manuscript.dvi
+manuscript/manuscript.log
+manuscript/manuscript.out
+manuscript/manuscript.pdf
```
Comment (jhollist) [inline .gitignore]: Do we want to ignore the rendered pdf?  Might be nice to include it with the repo.

Other than this comment, the rest looks good to me.

**Comment by naupaka** [comment]
I'll second leaving the pdf up - much easier to read through.

**Comment by snim2** [comment]
Is it better to leave the PDF in now, when it can get out of sync with the `.md`. or have someone force-commit the PDF once the manuscript has been finished?

**Comment by naupaka** [comment]
A fair point - I'm happy to defer to whatever others want to do.

File: .gitignore (hunk 1 above)
Comment (dlebauer) [inline .gitignore]: I agree, and as of last month, github renders pdf's in the browser https://github.com/blog/1974-pdf-viewing

**Comment by snim2** [comment]
Well, it seems that more people want the PDF in than want it out. In which case everyone who pushes has to check that the PDF is in sync, and everyone who reviews the writing has to check that the PDF they are reading is up to date.

To help with this I have added a file called `pre-commit` which is a git pre-commit hook that checks the timestamp of `manuscript.md` and `manuscript.pdf`. 

To use the pre-commit hook just put it in the directory `$REPO/.git/hooks` and make sure it is executable. The file is a BASH script; I'm not sure if that is the best option as it isn't cross-platform. I'm happy to re-write it as a Python script if people prefer.

**Comment by jhollist** [comment]
I'm in agreement with @naupaka.  You make a good case for leaving it out.  

For the writing stage, easy enough for us to render locally.  When we have a submission, we can render and push that separately.

That being said, I haven't used hooks much, so I will look at what you wrote when I have the time!

**Comment by snim2** [comment]
OK, well, I think as a matter of principle it is better for people not to merge their own PRs, so I'll leave you guys to merge this one, or not. If you do choose to remove the `.pdf` then I don' think it should be too hard to do that by removing a couple of commits (git wiling!).

**Pull request merged**